src/Domain/Creators/RegisterDerivedWithCharm.cpp

```diff
@@ -20,69 +20,68 @@
 namespace domain {
 namespace creators {
 namespace DomainCreators_detail {
-using Affine = CoordinateMaps::Affine;
-using Affine2D = CoordinateMaps::ProductOf2Maps<Affine, Affine>;
-using Affine3D = CoordinateMaps::ProductOf3Maps<Affine, Affine, Affine>;
-using Equiangular = CoordinateMaps::Equiangular;
-using Equiangular2D = CoordinateMaps::ProductOf2Maps<Equiangular, Equiangular>;
+// alias to CoordinateMaps namespace:
+namespace Maps = CoordinateMaps;
+
+// aliases to individual CoordinateMaps:
+using Affine = Maps::Affine;
+using Affine2D = Maps::ProductOf2Maps<Affine, Affine>;
+using Affine3D = Maps::ProductOf3Maps<Affine, Affine, Affine>;
+using Equiangular = Maps::Equiangular;
+using Equiangular2D = Maps::ProductOf2Maps<Equiangular, Equiangular>;
 using Equiangular3D =
-    CoordinateMaps::ProductOf3Maps<Equiangular, Equiangular, Equiangular>;
+    Maps::ProductOf3Maps<Equiangular, Equiangular, Equiangular>;
 using Equiangular3DPrism =
-    CoordinateMaps::ProductOf3Maps<Equiangular, Equiangular, Affine>;
-using EquatorialCompression = CoordinateMaps::EquatorialCompression;
-using Wedge2D = CoordinateMaps::Wedge2D;
-using Wedge3D = CoordinateMaps::Wedge3D;
-using Wedge3DPrism = CoordinateMaps::ProductOf2Maps<Wedge2D, Affine>;
-using Identity2D = CoordinateMaps::Identity<2>;
-using Translation3D = CoordinateMaps::ProductOf2Maps<Affine, Identity2D>;
+    Maps::ProductOf3Maps<Equiangular, Equiangular, Affine>;
+using EquatorialCompression = Maps::EquatorialCompression;
+using Frustum = Maps::Frustum;
+using Wedge2D = Maps::Wedge2D;
+using Wedge3D = Maps::Wedge3D;
+using Wedge3DPrism = Maps::ProductOf2Maps<Wedge2D, Affine>;
+using Identity2D = Maps::Identity<2>;
+using Translation3D = Maps::ProductOf2Maps<Affine, Identity2D>;
+using DiscreteRotation1D = Maps::DiscreteRotation<1>;
+using DiscreteRotation2D = Maps::DiscreteRotation<2>;
+using DiscreteRotation3D = Maps::DiscreteRotation<3>;
+
+template <typename... Maps>
+using LGMap = CoordinateMap<Frame::Logical, Frame::Grid, Maps...>;
+
+template <typename... Maps>
+using LIMap = CoordinateMap<Frame::Logical, Frame::Inertial, Maps...>;
 
 template <size_t Dim>
 void register_with_charm();
 
 template <>
 void register_with_charm<1>() {
```

Comment:
I think all of this could be simplified using:
```cpp
struct RegisterImpl {
  template <typename Map>
  void operator()(tmpl::type_<Map> /*meta*/) noexcept {
    PUPable_reg(SINGLE_ARG(Map));
  }
};

template <>
void register_with_charm<1>() {
  tmpl::for_each<tmpl::list<LIMap<Affine>, LIMap<DiscreteRotation1D, Affine>>(
    RegisterImpl{});
}
```
and similar `for_each` for 2d and 3d

Reply:
Yes, something like that, but also having the `tmpl::list`s automatically constructed by looping over the `DomainCreator<VolumeDim, TargetFrame>::creatable_classes` and having each `DomainCreator` specify a `coordinate_maps` type list.  This would get rid of almost all of these aliases and make registration more automatic.  Of course, I am not sure how any of this interacts with time-dependent maps...

Reply:
I see, yes, that can be done quite easily. So here's what I'm thinking. Let's take `Brick` as an example. It has as the derived class coordinate `CoordinateMaps::ProductOf3Maps<Affine, Affine, Affine>`. In the hpp the class can then have a type alias:
```cpp
using coordinate_maps = tmpl::list<CoordinateMaps::ProductOf3Maps<Affine, Affine, Affine>>>;
```
and in the `cpp` there should be an `ASSERT` that the result of `make_vector_coordinate_map_base<Frame::Logical, TargetFrame>` can be downcast to the type in the list. e.g.
```cpp
ASSERT(dynamic_cast<const tmpl::front<coordinate_maps>* const>(&coordinate_map_base) != nullptr,
       "words");
```

If the above change is made to the domain creators then the registration code can do what @kidder suggested:
```cpp
namespace {
struct RegisterImpl {
  template <typename Map>
  void operator()(tmpl::type_<Map> /*meta*/) const noexcept {
    PUPable_reg(SINGLE_ARG(Map));
  }
};

template <typename T>
struct get_maps_list {
  using type = typename T::coordinate_maps;
};
}  // namespace

template <size_t Dim>
void register_with_charm() noexcept {
  using all_maps = tmpl::remove_duplicates<tmpl::flatten<tmpl::transform<
    typename DomainCreator<Dim, Frame::Inertial>::createable_classes, get_maps_list<tmpl::_1>>>>;
  tmpl::for_each<all_maps>(RegisterImpl{});
}

template register_with_charm<1>() noexcept;
template register_with_charm<2>() noexcept;
template register_with_charm<3>() noexcept;
```

It'll be reasonably straight forward to do a loop over all `Frame`s as well, if that's ever necessary.

Some notes on the design:
- having `coordinate_maps` by a `tmpl::list` as opposed to just a map is done anticipating that when we have time-dependent maps we will need to register more than one map per DomainCreator.
- The instantiations could be done with `GENERATE_INSTANTIATIONS`, what I have was easier to type on GitHub :)

Reply:
LGTM